### PR TITLE
honor context when checking shortcut

### PR DESCRIPTION
**Pull request by joesis**
For https://github.com/getlantern/lantern-internal/issues/2564

### CODE REVIEW

File: shortcut.go [hunk 1]
```diff
@@ -46,20 +48,23 @@ func New(ipv4Subnets []string, ipv6Subnets []string) Shortcut {
 	return &shortcut{
 		v4list: newSortList(ipv4Subnets),
 		v6list: newSortList(ipv6Subnets),
+		// Prefers the system resolver by default, hopefully can use OS DNS cache.
+		resolver: net.DefaultResolver,
```
Comment (oxtoacart) [inline shortcut.go]: The type `Resolver` has a `PreferGo` setting that can be used to force a Go implementation of the DNS lookup (which does not cache). This is `false` by default and on Windows at least it's using syscalls to the system's resolver, so we should be fine on caching.

**Comment by oxtoacart** [comment]
LGTM!

File: shortcut.go (hunk 1 above)
Comment (joesis) [reply shortcut.go]: Yeah verified on macOS that it uses cache too.

**Comment by joesis** [comment]
Thanks @oxtoacart !

**Pull request merged**